tests/test_foreign.py

```diff
@@ -85,6 +85,14 @@ def get_list_of_lists(result):
         result = list(prolog.query("get_list_of_lists(Result)"))
         self.assertTrue({'Result': [[1], [2]]} in result, 'Nested lists should be unified correctly as return value.')
 
+    def test_dictionary(self):
```

Comment:
Are more test cases necessary? Like an empty dictionary?

Reply:
Added test cases for an empty dictionary and a nested dictionary